src/modules/universal-discovery/components/content-meta-preview/content.meta.preview.component.js

```diff
@@ -72,13 +88,14 @@ export default class ContentMetaPreviewComponent extends Component {
      */
     renderSelectContentBtn() {
         const {data, canSelectContent, onSelectContent, labels} = this.props;
-        const canSelect = canSelectContent(data.ContentInfo.Content);
         const attrs = {
             className: 'c-meta-preview__btn--select',
             onClick: onSelectContent
         };
 
-        if (!canSelect) {
+        canSelectContent(data, this.toggleDisableState.bind(this));
```

Comment:
I'm still trying to make up an another solution for it. It will run the checking everytime the button is rerendered, but the following code will be executed without waiting fo the results of that check.
I think it might cause issues in the future, because there might be a racing condition.
The button should be rendered after the `canSelectContent` method's callback is invoked, not earlier.

Reply:
The button will be disabled until the `canSelectContent` method will allow enabling the button.